Source/Core/VideoBackends/OGL/TextureConverter.cpp

```diff
@@ -241,12 +241,12 @@ static void EncodeToRamUsingShader(GLuint srcTexture,
 	// .. and then read back the results.
 	// TODO: make this less slow.
 
-	int writeStride = bpmem.copyMipMapStrideChannels * 32;
+	//int writeStride = bpmem.copyMipMapStrideChannels * 32;
 	int dstSize = dstWidth*dstHeight*4;
-	int readHeight = readStride / dstWidth / 4; // 4 bytes per pixel
-	int readLoops = dstHeight / readHeight;
+	//int readHeight = readStride / dstWidth / 4; // 4 bytes per pixel
+	//int readLoops = dstHeight / readHeight;
 
-	if (writeStride != readStride && readLoops > 1)
+	if (dstStride != dstWidth)
```

Comment:
```
pbo += dstWidth * 4;
destAddr += dstStride*2;
```
```
if (dstStride != dstWidth)
```
One of those blocks is wrong. In fact, I like the old if case more:
```
if (dstStride != srcStride && readLoops > 1)
```
So we can reuse srcStride for any dstWidth * 4.